### PR TITLE
fix hightlighted lines disappear or jump around

### PR DESCRIPTION
**Pull request by tom-shan**
#### What it does

fix https://github.com/eclipse-theia/theia/issues/8522
when theia receives the "thread" event which is generated by
another debug session, theia will change to that session see

#### How to test

debug the code below, add a breakpoint to line 36, see if the hightlighted yellow line will disappear or jump around
```py
from multiprocessing import Process
import threading
import time

class myThread (threading.Thread):
   def __init__(self, threadID, name, counter):
      threading.Thread.__init__(self)
      self.threadID = threadID
      self.name = name
      self.counter = counter
   def run(self):
      print_time(self.name, 10, self.counter)

def print_time(threadName, counter, delay):
   while counter:
      time.sleep(delay)
      counter -= 1

def print_func(continent='Asia'):
    thread1 = myThread(1, "Thread-1", 1)
    thread1.start()
    time.sleep(120)
    print('The name of continent is : ', continent)

if __name__ == "__main__":  # confirms that the code is under main function
    names = ['America', 'Europe', 'Africa']
    procs = []
    proc = Process(target=print_func)  # instantiating without any argument
    procs.append(proc)
    proc.start()

    # instantiating process with arguments
    for name in names:
        proc = Process(target=print_func, args=(name,))
        procs.append(proc)
        proc.start()

    # complete the processes
    for proc in procs:
        proc.join()
````
#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/debug/src/browser/debug-session.tsx [hunk 1]
```diff
@@ -221,12 +221,12 @@ export class DebugSession implements CompositeTreeElement {
     get currentThread(): DebugThread | undefined {
         return this._currentThread;
     }
-    set currentThread(thread: DebugThread | undefined) {
+    setCurrentThread(forceUpdateSession: boolean, thread: DebugThread | undefined): void {
```
Comment (vince-fugnitto) [inline packages/debug/src/browser/debug-session.tsx:224]: This is a breaking change, if possible please try not to break the API, else a note will be required in the `changelog` for the next release in the `breaking changes` section.

File: packages/debug/src/browser/debug-session.tsx [hunk 2]
```diff
@@ -414,31 +414,31 @@ export class DebugSession implements CompositeTreeElement {
...
         }
-        this.updateCurrentThread();
+        this.updateCurrentThread(forceUpdateSession);
     }
 
-    protected readonly scheduleUpdateThreads = debounce(() => this.updateThreads(undefined), 100);
+    protected readonly scheduleUpdateThreadsWithoutUpdateSession = debounce(() => this.updateThreads(false, undefined), 100);
     protected pendingThreads = Promise.resolve();
-    updateThreads(stoppedDetails: StoppedDetails | undefined): Promise<void> {
+    updateThreads(forceUpdateSession: boolean, stoppedDetails: StoppedDetails | undefined): Promise<void> {
```
Comment (vince-fugnitto) [inline packages/debug/src/browser/debug-session.tsx:429]: This is also a breaking change, any downstream application that calls this method will be broken by the changes.
In order to not break the API, you can use optional parameters.

**Review by vince-fugnitto — CHANGES_REQUESTED** [review]

File: packages/debug/src/browser/debug-session.tsx (hunk 2 above)
Comment (tom-shan) [reply packages/debug/src/browser/debug-session.tsx]: I have set a default true to forceUpdateSession

File: packages/debug/src/browser/debug-session.tsx (hunk 1 above)
Comment (tom-shan) [reply packages/debug/src/browser/debug-session.tsx]: I have wrapped setCurrentThread in currentThread.

**Comment by vince-fugnitto** [comment]
Sorry for the late response I hadn't seen the ping. 
Do you mind rebasing your pull-request so we can proceed with another round of review?

**Comment by tom-shan** [comment]
never mind, I have rebased it and resolved the conflicts, please review again

**Comment by tom-shan** [comment]
the check is unstable, rerun the check

**Comment by JonasHelming** [comment]
@vince-fugnitto Could you have a look and decide what to do with this PR, please?

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto Could you have a look and decide what to do with this PR, please?

@JonasHelming I'm just not a fan of having to pass a boolean around all the callers in order to get the fix to work, and haven't had the time to investigate a cleaner fix :(

**Pull request closed without merging**